src/components/Modal.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import tw, { styled } from "twin.macro";
+
+import Portal from "./Portal";
+import { Times } from "../assets/icons";
+
+type ModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const ModalComponent: React.ForwardRefRenderFunction<
+  HTMLElement,
+  React.PropsWithChildren<ModalProps>
+> = ({ children, isOpen, onClose }, externalRef) => {
+  if (!isOpen) {
+    return null;
+  }
+  return (
+    <Portal>
+      <Wrapper ref={externalRef}>
+        <ExitButton onClick={onClose}>
+          <Times />
+        </ExitButton>
+        <Content>{children}</Content>
+      </Wrapper>
+      <BgBlur />
+    </Portal>
+  );
+};
+
+const Modal = React.forwardRef(ModalComponent);
+Modal.displayName = "Modal";
+export default Modal;
+
+export const Wrapper = styled.aside`
```

Comment:
what is this wrapper class meant to be? is it like a card or something

Reply:
yep, it's the modal card